### ISSUE TITLE
localhost fails on OS X Sonoma / Safari 17

### ISSUE THREAD

**Issue by Krinkle** [browserstack/browserstack-runner#263]
I'm experiencing intermittent failures where tests on OS X Sonoma with Safari 17.3 fail after a 10 minute timeout.

By logging into the BrowserStack Automate panel during, and using "Start interactive session" to understand what was happening, I noticed that it said

> http://localhost:8899/test/index.html?_worker_key=1270749d-231b-492d-b31e-278e3886f510&_browser_string=OS%20X%20Sonoma%2C%20Safari%2017.3 not reachable

Clicking on the address bar and changing the hostname to `bs-local.com:8899` showed a passing test. So it seems whatever has been affecting iOS is now affecting macOS as well?

**Comment by Prakash21k** [comment]
Yes this is expected, due to security restrictions in Safari, we would need to use the URL as bs-local instead of localhost
